### PR TITLE
Fix compilation with Visual C++ 2015

### PR DESCRIPTION
**Pull request by Supermario1313**

### CODE REVIEW

**Comment by Wohlstand** [comment]
Be sure to don't break compilation on other platforms. Gonna verify this as soon as I can.

**Comment by Wohlstand** [comment]
Okay, I took a look and the reason why I made these lines is that there are build failures happening at some platforms (I don't remind on which), and I made that:
https://github.com/WohlSoft/luabind-deboostified/commit/8274224071513bdc6ee4429bd7619654f4e4683d

You should add the `#ifdef _MSC_VER` macros to use the different variants of this.

File: luabind/out_value_policy.hpp [hunk 1]
```diff
@@ -197,7 +197,7 @@ namespace luabind {
 			template<class T>
 			T& to_cpp(lua_State*, by_reference<T>, int)
 			{
-				storage_.decltype(storage_)::template construct<T>();
+				storage_.template construct<T>();
```
Comment (Wohlstand) [inline luabind/out_value_policy.hpp]: Please use macros to don't break other builds:
```cpp
#ifdef _MSC_VER
    storage_.template construct<T>();
#else
    storage_.decltype(storage_)::template construct<T>();
#endif

File: luabind/out_value_policy.hpp [hunk 2]
```diff
@@ -218,7 +218,7 @@ namespace luabind {
 			template<class T>
 			T* to_cpp(lua_State*, by_pointer<T>, int)
 			{
-				storage_.decltype(storage_)::template construct<T>();
+				storage_.template construct<T>();
```
Comment (Wohlstand) [inline luabind/out_value_policy.hpp]: Same as above.

**Review by Wohlstand — CHANGES_REQUESTED** [review]

**Comment by Supermario1313** [comment]
Fixed. I still wonder what compiler has a problem with this syntax though. From what I've seen on the Internet, it's the standard way of calling a templated member function and it compiles without errors on MSVC and Clang.

**Comment by Wohlstand** [comment]
Don't forget about CLang on other OS and GCC (and various versions of all of them). MSVC is the most annoying because it sometimes a pain at templates features support. Additionally, MSVC fails to get a pointer to the sub-steucture's field when other compilers works just fine.

**Comment by Wohlstand** [comment]
Okay, I'll merge the thing when I get my PC.

**Review by Wohlstand — APPROVED** [review]
Should be fine now.

**Pull request merged**